lib/marker.ts

```diff
@@ -0,0 +1,24 @@
+export function addTsdiMarker(o: any): void {
+  Object.defineProperty(o, '__tsdi_marker__', {
+    value: Symbol(),
+    configurable: false,
+    enumerable: false,
+    writable: false
+  });
+}
+
+export function findTsdiMarker(o: any): any {
+  let proto = o;
+  while (proto !== null) {
+    if (Object.getOwnPropertyDescriptor(proto, '__tsdi_marker__')) {
```

Comment:
Because on the long run I want to get rid of `reflect-metadata`.
The compiler version does not rely on `emitDecoratorMetadata` anymore and therefore you don't need `reflect-metadata`. Which by the way is very heavyweight with 50k.